src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTableOverRecordType.cs

```diff
@@ -72,9 +72,17 @@ public bool IsThisRecord(ISymbolSlot slot)
         {
             return slot.SlotIndex == int.MaxValue;
         }
-
-        internal override bool TryLookup(DName name, out NameLookupInfo nameInfo)
+                
+        bool INameResolver.Lookup(DName name, out NameLookupInfo nameInfo, NameLookupPreferences preferences)
         {
+            if (preferences.HasFlag(NameLookupPreferences.GlobalsOnly))

```

Comment:
This is the key fix. 

We conveniently know which symbols are row scope because they're in this class.